lib/driver.js

```diff
@@ -484,6 +482,37 @@ export class XCUITestDriver extends BaseDriver {
     }
   }
 
+  async handleMjpegOptions() {
+    const mjpegServerPort = this.opts.mjpegServerPort || DEFAULT_MJPEG_SERVER_PORT;
+    await this.allocateMjpegServerPort(this.opts.udid,
+      mjpegServerPort, this.isRealDevice() ? true : false);
+    // turn on mjpeg stream reading if requested
+    if (this.opts.mjpegScreenshotUrl) {
+      this.log.info(`Starting MJPEG stream reading URL: '${this.opts.mjpegScreenshotUrl}'`);
+      this.mjpegStream = new mjpeg.MJpegStream(this.opts.mjpegScreenshotUrl);
+      await this.mjpegStream.start();
+    }
+  }
+
+  async allocateMjpegServerPort(udid, mjpegServerPort, usePortForwarding) {
+    this.log.debug(
+      `MJPEG broadcasting requested, forwarding MJPEG server port ${mjpegServerPort} ` +
+      `to local port ${mjpegServerPort}`
+    );
+    try {
+      await DEVICE_CONNECTIONS_FACTORY.requestConnection(udid, mjpegServerPort, {
+        devicePort: mjpegServerPort,
+        usePortForwarding,
+      });
+    } catch (error) {
+       throw new Error(
```

Comment:
no exception must be thrown if this.opts.mjpegServerPort is not provided explicitly (e.g. default port value is used)

Reply:
Even if default port is picked and it busy then we should throw right ?

Reply:
not really. The exception should be thrown if one excplitily provides the `mjpegServerPort` capability value. If that option is undefined then we just print a warning if port forwarding fails

Reply:
something like:

```
catch (error) {
  if (_.isUndefined(this.opts.mjpegServerPort)) {
     this.log.warn ...
  } else {
     throw new Error ...
  }
}
```